### PR TITLE
[CI] Separate API diff from the test comment to move running the tests in a diff agent.

### PR DESCRIPTION
**Pull request by mandel-macaque**
The new module is not yet used. This is a first step, review the code, next create a PR in which the module is used. We add a yaml to have a github action ensuring the pwsh tests are run, that way we at least have a way to validate the pwsh used in the azure pipeline without using the pipeline.

### CODE REVIEW

File: tools/devops/automation/scripts/APIDiff.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+          "message" = $message;
+        }
+    }
+
+    It "creates with all data" {
+        $diff = [APIDiff]::new($header, $apiDiffData)
+
+        $diff.Header | Should -Be $header 
+
+        # assert that the linsk are correct
```
Comment (tj-devel709) [inline tools/devops/automation/scripts/APIDiff.Tests.ps1:78]:
```suggestion
        # assert that the links are correct
```

File: tools/devops/automation/scripts/APIDiff.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        $apiDiffData = @{
+          "gist" = $gists;
+          "html" =  @{};
+          "result" = $result;
+          "message" = $message;
+        }
+
+        $diff = [APIDiff]::new($header, $apiDiffData)
+
+        # assert that the linsk are correct
```
Comment (tj-devel709) [inline tools/devops/automation/scripts/APIDiff.Tests.ps1:133]:
```suggestion
        # assert that the links are correct
```

File: tools/devops/automation/scripts/APIDiff.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        $apiDiffData = @{
+          "gist" = $gists;
+          "html" = $null 
+          "result" = $result;
+          "message" = $message;
+        }
+
+        $diff = [APIDiff]::new($header, $apiDiffData)
+
+        # assert that the linsk are correct
```
Comment (tj-devel709) [inline tools/devops/automation/scripts/APIDiff.Tests.ps1:149]:
```suggestion
        # assert that the links are correct
```

File: tools/devops/automation/scripts/APIDiff.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+        foreach($platform in $gists.Keys) {
+            $diff.Platforms.ContainsKey($platform) | Should -Be $true
+            $diff.Platforms[$platform].Gist | Should -Be $gists[$platform]
+            $diff.Platforms[$platform].Html | Should -Be $null
+        }
+    }
+}
+
+Describe "API difff comment tests" {
```
Comment (tj-devel709) [inline tools/devops/automation/scripts/APIDiff.Tests.ps1:158]:
```suggestion
Describe "API diff comment tests" {
```

File: tools/devops/automation/scripts/APIDiff.psm1 [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    [string] $Message
+    [Hashtable] $Platforms
+
+    APIDiff (
+        [string] $header,
+        [Hashtable] $content) {
+        # the content object has the following format:
+        # 
+        #     gist: Hastable with platform as a key and the gist as content
+        #     html: Hastable with platform as a jey and the html url as content
```
Comment (tj-devel709) [inline tools/devops/automation/scripts/APIDiff.psm1:34]:
```suggestion
        #     html: Hastable with platform as a key and the html url as content
```

**Review by tj-devel709 — APPROVED** [review]
Few misspellings but LGTM

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**